Assets/MixedRealityToolkit/_Core/Interfaces/IMixedRealityBoundarySystem.cs

```diff
@@ -36,15 +37,45 @@ public interface IMixedRealityBoundarySystem : IMixedRealityManager
         bool EnablePlatformBoundaryRendering { get; set; }
 
         /// <summary>
-        /// A three dimensional volume as described by the smallest rectangle
-        /// containing the complete playspace and the configured height.
+        /// Two dimensional representation of the geometry of the boundary, as provided
+        /// by the platform.
         /// </summary>
-        Bounds OutscribedVolume { get; }
+        /// <remarks>
+        /// BoundaryGeometry should be treated as the outline of the player's space, placed
+        /// on the floor.
+        /// </remarks>
+        Edge[] GeometryBounds { get; }
+
+        /// <summary>
+        /// The largest rectangle that is contained withing the playspace geometry.
+        /// </summary>
+        InscribedRectangle InscribedRectangularBounds { get; }
 
         /// <summary>
-        /// A three dimensional volume as described by the largest rectangle that
-        /// is contained withing the playspace and the configured height.
+        /// Indicates the height of the floor, in relation to the coordinate system origin.
         /// </summary>
-        Bounds InscribedVolume { get; }
+        /// <remarks>
+        /// If a floor has been located, FloorHeight.HasValue will be true, otherwise it will be false.
+        /// </remarks>
+        float? FloorHeight { get; }
```

Comment:
Does this need to be a nullable float?  Surely 0 would suffice without the overhead. (or -1)

Reply:
0 and -1 are valid `FloorHeight`s, since it exists along the Y axis.

Reply:
There's not THAT much overhead and -1, in theory is a valid floor offset. I'd rather keep the nullable so that we can know that no floor was found

Reply:
I think nullable makes sense here.